recipes/theora/all/CMakeLists.txt

```
@@ -0,0 +1,130 @@
+  "include/theora/codec.h"
+  "include/theora/theora.h"
+  "include/theora/theoradec.h"
+  "include/theora/theoraenc.h"
+)
+
+set(LIBTHEORA_COMMON
+   "lib/apiwrapper.c"
+   "lib/bitpack.c"   # was not in enc in SConstruct
+   "lib/dequant.c"
+   "lib/fragment.c"
+   "lib/idct.c"
+   "lib/info.c"   # was not in enc in SConstruct
+   "lib/internal.c"
+   "lib/state.c"
+   "lib/quant.c"
+   )
+
+set(LIBTHEORA_ENC
+   "lib/analyze.c"
+   "lib/encapiwrapper.c"
+   "lib/encfrag.c"
+   "lib/encinfo.c"
+   "lib/encode.c"
+   "lib/enquant.c"
+   "lib/fdct.c"
+   "lib/huffenc.c"
+   "lib/mathops.c"
+   "lib/mcenc.c"
+   "lib/rate.c"
+   "lib/tokenize.c"
+   )
+
+
+set(LIBTHEORA_DEC
+   "lib/decapiwrapper.c"
+   "lib/decinfo.c"
+   "lib/decode.c"
+   "lib/huffdec.c"
+   )
+
+
+option(USE_X86_32 "Use x86-32 optimization" OFF)
+option(USE_X86_64 "Use x86-64 optimization" OFF)
+if (USE_X86_32)
+   message("Enabling x86-32 assembly optimizations")
+   add_definitions(-DOC_X86_ASM)
+
+   list (APPEND LIBTHEORA_COMMON
+      "lib/x86/mmxidct.c"
+      "lib/x86/mmxfrag.c"
+      "lib/x86/mmxstate.c"
+      "lib/x86/x86state.c"
+      )
+
+   list (APPEND LIBTHEORA_ENC
+      "lib/x86/mmxencfrag.c"
+      "lib/x86/mmxfdct.c"
+      "lib/x86/x86enc.c"
+      "lib/x86/mmxfrag.c"
+      "lib/x86/mmxidct.c"
+      "lib/x86/mmxstate.c"
+      "lib/x86/x86state.c"
+      )
+elseif (USE_X86_64)
+   message("Enabling x86-64 assembly optimizations")
+   add_definitions(-DOC_X86_ASM)
+   add_definitions(-DOC_X86_64_ASM)
+
+   list (APPEND LIBTHEORA_COMMON
+      "lib/x86/mmxidct.c"
+      "lib/x86/mmxfrag.c"
+      "lib/x86/mmxstate.c"
+      "lib/x86/x86state.c"
+      )
+
+   list (APPEND LIBTHEORA_ENC
+      "lib/x86/mmxencfrag.c"
+      "lib/x86/mmxfdct.c"
+      "lib/x86/x86enc.c"
+      "lib/x86/sse2fdct.c"
+      )
+else()
+   message("Not enabling any assembly optimizations (x86 and x86_64 asm optimizations are available)")
+endif()
+
+
+if (MSVC AND BUILD_SHARED_LIBS)
+   set (LIBTHEORA_DEF "lib/theora.def")
+   set (LIBTHEORA_ENC_DEF "win32/xmingw32/libtheoraenc-all.def")
+   set (LIBTHEORA_DEC_DEF "win32/xmingw32/libtheoradec-all.def")
+endif ()
+
+add_library(theora ${LIBTHEORA_COMMON} ${LIBTHEORA_ENC} ${LIBTHEORA_DEC} ${LIBTHEORA_DEF})
+add_library(theoraenc ${LIBTHEORA_COMMON} ${LIBTHEORA_ENC} ${LIBTHEORA_ENC_DEF})
+add_library(theoradec ${LIBTHEORA_COMMON} ${LIBTHEORA_DEC} ${LIBTHEORA_DEC_DEF})
+
+if (BUILD_SHARED_LIBS)
+   target_compile_definitions(theora PRIVATE LIBTHEORA_EXPORTS)
+   target_compile_definitions(theoraenc PRIVATE LIBTHEORA_EXPORTS)
+   target_compile_definitions(theoradec PRIVATE LIBTHEORA_EXPORTS)
+endif()
+
+if (MSVC)
+   target_compile_definitions (theora PRIVATE _CRT_SECURE_NO_DEPRECATE _CRT_NONSTDC_NO_DEPRECATE)
+   target_compile_definitions (theoraenc PRIVATE _CRT_SECURE_NO_DEPRECATE _CRT_NONSTDC_NO_DEPRECATE)
+   target_compile_definitions (theoradec PRIVATE _CRT_SECURE_NO_DEPRECATE _CRT_NONSTDC_NO_DEPRECATE)
+endif ()
```

No need to pollute the code, it's just warnings

```suggestion
endif()
```